Automattic/jetpack
# JITM: allow the pre-connection JITM icon to be set

#### Changes proposed in this Pull Request:
Allow consumer plugins to select the icon to be used in the pre-connection JITMs.

* Use `jetpack` by default if the `icon` key does not exist in the pre-connection JITM array.
* Update the readme with info on setting the icon.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install and activate this branch. Do not connect Jetpack.
2. Add a pre-connection using a snippet like the one shown below. You can use the Code Snippets plugin to do this if you'd like.
```
function add_preconnection_jitms( $messages ) {
    $example_jitm = array(
		'id'             => 'example-jitm',
		'message_path'   => '/wp:plugins:admin_notices/',
		'message'        => __( 'An example message.', 'jetpack' ),
		'description'    => __( 'An example description.', 'jetpack' ),
		'button_link'    => 'https://example.com/path',
		'button_caption' => __( 'Example button text', 'jetpack' ),
		'icon'           => '',
    );

    if ( ! is_array( $messages ) ) {
		return array( $example_jitm );
    }

    return array_merge( $messages, array( $example_jitm ) );
 }

 add_filter( 'jetpack_pre_connection_jitms', 'add_preconnection_jitms' );
```
3. Navigate to the page where the pre-connection JITM is displayed (the plugins page for the JITM above). Verify that no icon is displayed in the JITM.
4. Update the snippet to use 'jetpack' or 'woocommerce' for the icon, and verify that the correct icon is displayed in the JITM.
5. Remove the 'icon' key from the JITM array and verify that the Jetpack icon is displayed.
